torch/nn/parallel/parallel_apply.py

```diff
@@ -8,31 +8,37 @@
     import Queue as queue
 
 
-def parallel_apply(modules, inputs):
+def parallel_apply(modules, inputs, kwargs_tup=None):
     assert len(modules) == len(inputs)
+    if kwargs_tup:
+        assert len(modules) == len(kwargs_tup)
+    else:
+        kwargs_tup = tuple({} for gpu in modules)
     # Fast track
     if len(modules) == 1:
-        return (modules[0](*inputs[0]),)
+        return (wrap(modules[0], *inputs[0], **kwargs_tup[0]), )
 
     lock = threading.Lock()
     results = {}
 
-    def _worker(module, input, results, lock):
+    def _worker(module, input, results, lock, **kwargs):
```

Comment:
I think it's better to pass the kwargs as a single dict argument (probably next to the input).